packages/venia-ui/lib/drivers/adapter.js

```diff
@@ -46,44 +47,37 @@ const storage = new BrowserPersistence();
  */
 const VeniaAdapter = props => {
     const { apiBase, apollo = {}, children, store } = props;
+    const [initialized, setInitialized] = useState(false);
 
-    const cache = apollo.cache || preInstantiatedCache;
-    const link = apollo.link || VeniaAdapter.apolloLink(apiBase);
-
-    const persistor = new CachePersistor({
-        cache,
-        storage: window.localStorage,
-        debug: process.env.NODE_ENV === 'development'
-    });
-
-    let apolloClient;
-    if (apollo.client) {
-        apolloClient = apollo.client;
-    } else {
-        apolloClient = new ApolloClient({
+    const apolloClient = useMemo(() => {
```

Comment:
**This should always have been memoized.** Performing side effects in the body of a render function (component) has always been an anti-pattern; here, the original code was creating multiple Apollo clients as `Adapter` is executed several times during initial rendering.

An easy mistake to make, but a very important one to avoid.